server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RemoteDelivery.java

```diff
@@ -266,12 +255,14 @@ public void service(Mail mail) throws MessagingException {
      */
     @Override
     public synchronized void destroy() {
-        volatileIsDestroyed.markAsDestroyed();
-        // Wake up all threads from waiting for an accept
-        for (Thread t : workersThreads) {
-            t.interrupt();
+        if (startThreads) {
```

Comment:
why don't use Executor abstraction for worker threads ? It would be much simpler